.github/workflows/build_and_test.yaml

```diff
@@ -28,6 +28,13 @@ jobs:
     - uses: ./tools/github-actions/setup-deps
     - run: make -k gen-check
 
+  license-check:
```

Comment:
nit: if we have more of these, we should consider creating `check` and add `licencse-check` and `gem-check` under it

Reply:
Yes, I went back and forth on breaking this out, but decided three should be the critical number. Does that seem okay?

Reply:
sg thanks for sharing the thought process

Reply:
On the other side, I'd argue that this should be part of the existing `make lint`.

Reply:
`boilerplate.py` has been removed, so this target has also been removed.

Reply:
I'm going to leave this as-is for now, so that we can have proper licensing for our first big public release. I'm happy for targets to move around at a later date though.

Reply:
Since we're at the v0.2.0 release date, we can iterate to improve in v0.3.0.